### ISSUE TITLE
Training Set Format

### ISSUE THREAD

**Issue by GXYM** [sudban3089/ID-Preserving-Facial-Aging#2]
I am very interested in your work, but I am not very familiar with the format of Training Set. Can you tell me how to organize the data format of Training Set For example, everyone's images should be placed in the same folder, and different people should be placed in different folders; Do you have any special requirements for the name of the picture?

**Comment by sudban3089** [comment]
Thanks for your interest in our work. Please refer to IJCB2023_TrainExamples folder on the github page. We noticed some google drive access issue so we put example images in that folder for learning purposes. You will see in that folder there are subfolders where each subfolder (SubID_XXXX_NumImages_YY) contains images of a single individual of subject ID XXXX. Note there is no special requirement for filename for Training Set. We used the convention to help perform biometric matching.

**Comment by GXYM** [comment]
> Thanks for your interest in our work. Please refer to IJCB2023_TrainExamples folder on the github page. We noticed some google drive access issue so we put example images in that folder for learning purposes. You will see in that folder there are subfolders where each subfolder (SubID_XXXX_NumImages_YY) contains images of a single individual of subject ID XXXX. Note there is no special requirement for filename for Training Set. We used the convention to help perform biometric matching.

I have read your paper. I have some questions. Do I need to train a model for each independent person（for SubID_XXXX_NumImages_YY）. Or can I mix photos of different people and ages together during training. During testing, the images were also random.

**Comment by GXYM** [comment]
In readme "This repo currently supports single subject-specific training" what means?
Or what I want to ask is, which set of directory (a or b) should data_root be given during training.
a: Data_root=xxx/IJCB2023/TrainExamples
b: Data_root=xxx/IJCB2023/TrainExamples/SubID_2029NumImages_19
When testing, must the characters be existing in the training set

**Comment by sudban3089** [comment]
During our training we used images of a single individual for subject-specific training. Please note this is based on how DreamBooth works which fine-tunes on images of a specific instance (in our case, a single individual). There are currently ways to perform multi-concept fine-tuning, which you can explore but that is beyond the scope of our work. For your second qs, yes the data_root should be option 'b'. Testing is done for trained subjects. DreamBooth is used to learn facial characteristics of a particular individual using identity-preserving loss and the regularization set was used to make the model learn the relationship between how facial appearance changes with age which is used during inference "photo of a [token] person as [age label]".

**Comment by GXYM** [comment]
> For your second qs, yes the data_root should be option 'b'. Testing is done for trained subjects. DreamBooth is used to learn facial characteristics of a particular individual using identity-preserving loss and the regularization set was used to m

Thank you for your reply Only fine-tuning and reasoning for a single character greatly limits the application of the model. Thank you again for your excellent work, which has been very inspiring for my research.

**Comment by sudban3089** [comment]
We used ProFusion and other GAN-based methods that use single image. However, when the biometric evaluation and age prediction was done on the images age-edited using those methods, we observed subject-specific finetuning demonstrated better performance. As for the single user, I believe currently there are techniques that can learn images of multiple individuals simultaneously using tokens. Hope this helps.

**Comment by GXYM** [comment]
> We used ProFusion and other GAN-based methods that use single image. However, when the biometric evaluation and age prediction was done on the images age-edited using those methods, we observed subject-specific finetuning demonstrated better performance. As for the single user, I believe currently there are techniques that can learn images of multiple individuals simultaneously using tokens. Hope this helps.

Another question is whether we can choose one of the three different loss training scripts provided in readme, or do we need to go through three different loss training scripts in sequence?

**Comment by sudban3089** [comment]
You can choose one of the training loss scripts, we observe contrastive loss to yield best results. Please refer to the paper for comparison.

**Comment by sudban3089** [comment]
I am closing this issue as your original question on Training images has been resolved.
